library/src/main/java/kwasm/format/text/Type.kt

```diff
@@ -14,47 +14,57 @@
 
 package kwasm.format.text
 
-import kwasm.ast.ValueType
 import kwasm.format.ParseContext
 import kwasm.format.ParseException
+import kwasm.format.shiftColumnBy
 
+/**
+ * This sealed class encapsulates all Types defined in
+ * [the docs](https://webassembly.github.io/spec/core/text/types.html)
+ *
+ * @param T the Object representing the value related to the type and value parsed
+ */
 sealed class Type<T>(
     protected val sequence: CharSequence,
     protected val context: ParseContext? = null
 ) {
-    val value: T by lazy {
-        parseValue()
-    }
+    val value: T by lazy { parseValue() }
 
     protected abstract fun parseValue(): T
-
+    /**
+     * Encapsulates kwasm.ast.ValueType for parsing. Parses the sequence in accordance
+     * to the definition in [the docs](https://webassembly.github.io/spec/core/text/types.html#value-types)
+     *
+     * @constructor Parses the sequence passed in and populates the appropriate parsed value.
+     * @throws ParseException when the sequence falls outside of the spec definition
+     */
     class ValueType(
         sequence: CharSequence,
         context: ParseContext? = null
     ) : Type<kwasm.ast.ValueType>(sequence, context) {
-        override fun parseValue(): kwasm.ast.ValueType {
-            return when (sequence){
+        override fun parseValue(): kwasm.ast.ValueType = when (sequence){
                 "i32" -> kwasm.ast.ValueType.I32
                 "i64" -> kwasm.ast.ValueType.I64
                 "f32" -> kwasm.ast.ValueType.F32
                 "f64" -> kwasm.ast.ValueType.F64
-                else -> {
-                    throw ParseException("Invalid ValueType", context)
-                }
+                else -> throw ParseException("Invalid ValueType", context)
             }
         }
-    }
 
     class ResultType(
         sequence: CharSequence,
         context: ParseContext? = null
     ) : Type<kwasm.ast.ValueType?>(sequence, context) {
         override fun parseValue(): kwasm.ast.ValueType? {
+            if(sequence=="") {
```

Comment:
nit: spaces between `==` and its operands, also, it could be written as:

```kotlin
if (sequence.isEmpty()) return
```

Reply:
Done